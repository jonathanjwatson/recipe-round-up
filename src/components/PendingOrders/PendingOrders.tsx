import React, { useEffect, useState } from "react";
import API from "../../API/API";

const PendingOrders: React.FC = () => {
  const [orders, setOrders] = useState([
    {
      pending: false,
      recipe: 0
    }
  ]);

  useEffect(() => {
    API.getOrders()
      .then(response => {
        setOrders(response.data.orders);
      })
      .catch(err => {
        console.log(err);
      });
  }, []);

  return (
    <div>
      <h1>Pending Orders</h1>
      <table className="table">
        <thead className="thead-dark">
          <tr>
            <th scope="col">Order Index</th>
            <th scope="col">Recipe Number</th>
            <th scope="col">Status</th>
          </tr>
        </thead>
        <tbody>
          {orders.map((order, i: number) => (
            <tr key={i}>
              <th scope="row">{i}</th>
              <th scope="row">{order.recipe}</th>
              {order.pending && <td>Pending</td>}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default PendingOrders;
