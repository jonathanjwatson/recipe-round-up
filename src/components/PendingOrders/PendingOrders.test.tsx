import React from "react";
import { render, cleanup } from '@testing-library/react';
import PendingOrders from "./PendingOrders";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<PendingOrders />);
  expect(asFragment()).toMatchSnapshot();
});
