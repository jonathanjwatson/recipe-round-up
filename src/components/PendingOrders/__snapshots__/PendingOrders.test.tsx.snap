// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <h1>
    Pending Orders
  </h1>
  <table
    class="table"
  >
    <thead
      class="thead-dark"
    >
      <tr>
        <th
          scope="col"
        >
          Order Index
        </th>
        <th
          scope="col"
        >
          Recipe Number
        </th>
        <th
          scope="col"
        >
          Status
        </th>
        <th
          scope="col"
        >
          Total Times Made
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        />
        <th
          scope="row"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
